xla/service/gpu/runtime/nccl_clique.h

```diff
@@ -112,18 +125,29 @@ struct NcclCliqueName {
   }
 };
 
-struct NcclClique : public Lockable<NcclCliqueCommunicators, NcclCliqueName> {
+class NcclClique : public Lockable<NcclCliqueCommunicators, NcclCliqueName> {
+ public:
   // We keep acquired cliques in a sorted container to guarantee that all
   // participants iterate over cliques in the same order.
   using AcquiredCliquesMap =
       absl::btree_map<NcclCliqueKey, std::shared_ptr<NcclClique::Lock>,
                       std::greater<NcclCliqueKey>>;
 
+  // Construct the lockable clique.
+  // Note that async errors can be checked without acquiring the lock.
+  // To get the lock-free reference to the communicators for the async
+  // error checks, the constructor intentionally leaks the reference
+  // to the communicators from an acquired lock.
   NcclClique(NcclCliqueKey clique_key, std::optional<NcclCliqueId> clique_id,
              absl::btree_map<int32_t, NcclApi::OwnedNcclComm> communicators)
-      : Lockable(std::move(clique_key), clique_id, std::move(communicators)) {}
+      : Lockable(std::move(clique_key), clique_id, std::move(communicators)),
+        async_error_checker_(Acquire()->GetChecker()) {}
 
   std::string DebugString() const;
+  absl::Status Check();
```

Comment:
Can you also briefly describe the contract for this method?

Reply:
Done. Also renamed the method.